File: Containermostwater.py
class Solution(object):
    def maxArea(self, heights):
        """
        :type height: List[int]
        :rtype: int
        """
        area=0
        l=0
        r=len(heights)-1
        while(l<=r):
            
            area=max(area,min(heights[l],heights[r])*(r-l) )
            if heights[l]>heights[r]:
                r=r-1
            else:
                l=l+1
        return area
    

File: ContainerMostWater.java
//tc = o(n)
//sc = o(1)

class Solution {
    public int maxArea(int[] height) {
        int left = 0;
        int right = height.length-1;
        int maxarea = Integer.MIN_VALUE;
        while(left < right){
            int area = Math.min(height[left],height[right])* (right-left);
            maxarea = Math.max(maxarea,area);
            if(height[left] < height[right]){
                left++;
            }
            else{
                right--;
            }
        }
        return maxarea;
    }
}